### PR TITLE
Explicitly set seccomp profile for nginx-ingress-controller

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**How to categorize this PR?**

/area open-source security
/kind bug

**What this PR does / why we need it**:
After https://github.com/gardener/gardener/pull/6741 it is possible to enable cluster-wide seccomp defaulting. Also there is the [seccomp defaulting webhook](https://github.com/gardener/gardener/pull/6450/files). These do not play well with the `nginx-ingress-controller`. The `nginx-ingress-controller` needs a customized seccomp profile which is not well documented (see [ref](https://github.com/kubernetes/ingress-nginx/issues/8640)). Also customized seccomp profiles are not trivial to distribute. Some options will be a daemonset or a security profiles [operator](https://github.com/kubernetes-sigs/security-profiles-operator). Adding the `CAP_SYS_ADMIN` capability [ref](https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h) can make it work, but this actually alters the provided seccomp profile by enabling additional syscalls. This is not ideal because the container also gains the capabilities of `SYS_ADMIN`. Read more on seccomp vs linux capabilities [here](https://security.stackexchange.com/questions/210400/difference-between-linux-capabities-and-seccomp).

This PR explicitly sets the seccomp profile for the `nginx-ingress-controller` to Unconfined. This is the also what the current default seccomp profile is in kubernetes.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
An issue causing the `nginx-ingress-controller` installed via the shoot's `nginx-ingress` addon to fail to start when cluster-wide seccomp defaulting is enabled is now fixed.
```

### CODE REVIEW

File: pkg/operation/botanist/component/nginxingress/nginxingress.go [hunk 1]
```diff
@@ -579,6 +583,9 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 	// Skipped until https://github.com/kubernetes/ingress-nginx/issues/8640 is resolved
 	// and special seccomp profile is implemented for the nginx-ingress
 	deploymentController.Spec.Template.Labels[resourcesv1alpha1.SeccompProfileSkip] = "true"
+	if k8sVersionGreaterEqual119 {
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/nginxingress/nginxingress.go]: Nit: Gardener supports only Seed clusters with K8s >= 1.20.

https://github.com/gardener/gardener/blob/98ce217fd6bc916e9f08fb0d63d6480ab13ea7ab/pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go#L199-L211

We can omit this version check and assume that the Seed's K8s version is >= 1.20.

**Review by ialidzhikov — COMMENTED** [review]
Minor suggestion, otherwise lgtm.

File: pkg/operation/botanist/component/nginxingress/nginxingress.go (hunk 1 above)
Comment (dimityrmirchev) [reply pkg/operation/botanist/component/nginxingress/nginxingress.go]: Yes, you are right! I will make the change

File: pkg/operation/botanist/component/nginxingress/nginxingress.go [hunk 2]
```diff
@@ -437,6 +437,7 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 								},
 								RunAsUser:                pointer.Int64(101),
 								AllowPrivilegeEscalation: pointer.Bool(true),
+								SeccompProfile:           &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeUnconfined},
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/nginxingress/nginxingress.go:440]: Q: As we now explicitly specify the `seccompProfile`, do we benefit from https://github.com/gardener/gardener/blob/ec9c4743a94a21ffaf0a1848d6094cdc381e1a4e/pkg/operation/botanist/component/nginxingress/nginxingress.go#L579-L581 (for the `DefaultSeccompProfile` feature gate case)? The webhook does not overwrite the `seccompProfile` when it is already specified: https://github.com/gardener/gardener/blob/8a7d7b08f54a005be62c6bbfca50f91bb15246cf/pkg/resourcemanager/webhook/seccompprofile/handler.go#L69-L72

Or do you prefer to keep this small optimization that will prevent the webhook to be called for the nginx-ingress Pods?
Anyways, I guess this question is not that important in the longterm as the plan forward for ManagedSeeds is to benefit from the kubelet's seccomp profile defaulting (for K8s >= 1.25).

**Review by ialidzhikov — COMMENTED** [review]
A final question.

File: pkg/operation/botanist/component/nginxingress/nginxingress.go (hunk 2 above)
Comment (dimityrmirchev) [reply pkg/operation/botanist/component/nginxingress/nginxingress.go:440]: I prefer to leave the label for couple of reasons:
1. The webhook will not be called
2. I can select pods by labels (kind of documentation)

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/approve

**Pull request merged**